crates/storage/src/lazy/mod.rs

```diff
@@ -148,16 +148,25 @@ where
 
     /// Try to read the `value` from the contract storage.
     ///
+    /// To successfully retrieve the `value`, the encoded `key` and `value`
+    /// must both fit into the static buffer together.
+    ///
     /// Returns:
     /// - `Some(Ok(_))` if `value` was received from storage and could be decoded.
-    /// - `Some(Err(_))` if the encoded length of `value` exceeds the static buffer size.
+    /// - `Some(Err(_))` if retrieving the `value` would exceed the static buffer size.
     /// - `None` if there was no value under this storage key.
     pub fn try_get(&self) -> Option<ink_env::Result<V>> {
-        let encoded_length: usize = ink_env::contains_contract_storage(&KeyType::KEY)?
+        let key_size = KeyType::KEY.encoded_size();
```

Comment:
I quite like the fully qualified invocation of `encoded_size` like in `mapping` https://github.com/paritytech/ink/blob/48ed3f853659704bcf5e3e60aa49cf1acbc0bf4d/crates/storage/src/lazy/mapping.rs#L269

`<Q as Encode>::encoded_size`

For readability.